assembly/src/parsers/mod.rs

```diff
@@ -160,10 +160,21 @@ fn parse_op_token(
         "cdropw" => stack_ops::parse_cdropw(span_ops, op),
 
         // ----- input / output operations --------------------------------------------------------
-        "push" => io_ops::parse_push(span_ops, op, num_proc_locals),
-        "pushw" => io_ops::parse_pushw(span_ops, op, num_proc_locals),
+        "push" => io_ops::parse_push(span_ops, op),
+        "sdepth" => io_ops::parse_env_ops(span_ops, op, num_proc_locals),
+        "locaddr" => io_ops::parse_env_ops(span_ops, op, num_proc_locals),
+
+        "mem_load" => io_ops::parse_mem_ops(span_ops, op),
+        "mem_loadw" => io_ops::parse_mem_ops(span_ops, op),
+        "mem_store" => io_ops::parse_mem_ops(span_ops, op),
+        "mem_storew" => io_ops::parse_mem_ops(span_ops, op),
+
+        "loc_load" => io_ops::parse_local_ops(span_ops, op, num_proc_locals),
+        "loc_loadw" => io_ops::parse_local_ops(span_ops, op, num_proc_locals),
+        "loc_store" => io_ops::parse_local_ops(span_ops, op, num_proc_locals),
+        "loc_storew" => io_ops::parse_local_ops(span_ops, op, num_proc_locals),
+
         "pop" => io_ops::parse_pop(span_ops, op, num_proc_locals),
-        "popw" => io_ops::parse_popw(span_ops, op, num_proc_locals),
         "loadw" => io_ops::parse_loadw(span_ops, op, num_proc_locals),
         "storew" => io_ops::parse_storew(span_ops, op, num_proc_locals),
```

Comment:
A couple comments here:

1. Why do we still keep `pop`, `loadw`, and `storew` options here? The intent was to replace them with new operations.
2. Instead of grouping operations by memory vs. locals, I'd probably group them by logical similarity. For example, we could have `parse_loadw()` method which would parse both `mem_loadw` and `loc_loadw` instructions. I think these operations are quite similar and would share a lot of parsing code. Same would apply to other pairs. The end result is that we could probably get rid of `io_ops/local_ops.rs` file entirely and have all memory-related parsers in `io_ops/mem_ops.rs`.

Regarding the last point, it might actually make sense to have just two parsers for memory and local operations: `parse_mem_read` and `parse_mem_write`. Signatures for these could look as follows:

```Rust
fn parse_mem_read(
    span_ops: &mut Vec<Operation>,
    op: &Token,
    num_proc_locals: u32,
    is_local: bool,
    is_single: bool,
) -> Result<(), AssemblyError>

fn parse_mem_write(
    span_ops: &mut Vec<Operation>,
    op: &Token,
    num_proc_locals: u32,
    is_local: bool,
    is_single: bool,
) -> Result<(), AssemblyError>
```